app/javascript/pages/account/components/EmailFieldset.tsx

```diff
@@ -3,43 +3,57 @@ import React from "react"
 import { ErrorOption, UseFormMethods } from "react-hook-form"
 import Fieldset from "./Fieldset"
 import { emailRegex } from "../../../util/accountUtil"
-import { AxiosError } from "axios"
+import { ErrorMessages } from "./ErrorSummaryBanner"
+import { ExpandedAccountAxiosError, getErrorMessage } from "./util"
+import { renderInlineMarkup } from "../../../util/languageUtil"
 
 const validateEmail = (email: string) => {
   return emailRegex.test(email)
 }
 
 export const handleEmailServerErrors =
-  (setError: (name: string, error: ErrorOption) => void, errorCallback: () => void) =>
-  (error: AxiosError) => {
+  (setError: (name: string, error: ErrorOption) => void, errorCallback?: () => void) =>
+  (error: ExpandedAccountAxiosError) => {
     if (error.response.status === 422) {
-      setError("email", { message: "email:generalFormat", shouldFocus: true })
+      if (
+        (error.response.data?.errors?.full_messages || []).includes("Email has already been taken")
+      ) {
+        setError("email", { message: "email:server:duplicate", shouldFocus: true })
```

Comment:
You can do a similar fix here, where you just return the object that is the second argument of setError - I don't see any reason we need to be passing setError into these functions